Sources/BSON/Array+BSONValue.swift

```diff
@@ -0,0 +1,23 @@
+import NIO
+
+/// An extension of `Array` to represent the BSON array type.
+extension Array: BSONValue where Element == BSON {
+    static var bsonType: BSONType { .array }
```

Comment:
all these need explicit acl's. It is a bit redundant because `BSONValue` is `internal` itself, but it's nice to have that information included here as well.